frontend/lib/CMakeLists.txt

```diff
@@ -45,10 +59,40 @@ add_subdirectory(types)
 add_subdirectory(uast)
 add_subdirectory(util)
 
-add_library(libdyno SHARED $<TARGET_OBJECTS:libdyno-obj>)
+add_library(libdyno SHARED $<TARGET_OBJECTS:libdyno-obj> $<TARGET_OBJECTS:git-sha-obj>)
+
 target_include_directories(libdyno PUBLIC
                            ${CHPL_MAIN_INCLUDE_DIR}
                            ${CHPL_INCLUDE_DIR})
+
 target_link_libraries(libdyno PUBLIC ${CHPL_LLVM_LINK_ARGS})
 # TODO: Get printchplenv output proper so that we don't need to SHELL here
 target_compile_options(libdyno PUBLIC SHELL:$<$<COMPILE_LANGUAGE:CXX>:${CHPL_LLVM_COMP_ARGS}>)
+
+# TODO: the ideal way to cleanup would be to add ${CMAKE_CURRENT_SOURCE_DIR}/util/git-version.cpp
+# to the list of BYPRODUCTS, but that creates a strong cycle between libdyno and git-sha-obj
+# which is not allowed when the libraries are not static. The workaround so far is to set
+# the ADDITIONAL_CLEAN_FILES property of libdyno to add git-version.cpp, but that property
+# wasn't added until 3.15 and our reqs list 3.13.4
+# generate the git-version.cpp and compile it to produce the .o before linking libdyno
+# If .git doesn't exist, we are probably in a release tarball or Dockerfile
+# and expect that the git-version.cpp file was copied into our source tree
+# from their respective build process
+if (EXISTS ${CMAKE_SOURCE_DIR}/.git)
+  add_custom_command(TARGET libdyno
+                    PRE_LINK
+                    COMMAND ${CMAKE_SOURCE_DIR}/util/config/write-git-sha ${CMAKE_CURRENT_SOURCE_DIR}/util --build-version --chpl-home=${CHPL_HOME} # generate the file with git sha
```

Comment:
not sure if functions defined in `CMakeLists.txt` can be called in the `COMMAND`. It looks like you can call something defined by `add_executable` or `configure_file`, but otherwise it's assumed to be a call to something found in the system's path